### ISSUE TITLE
Clicking "Source" to open haddock source on hackage opens invalid URL

### ISSUE THREAD

**Issue by jhrcek** [haskell/haskell-language-server#2508]
### Your environment

Which OS do you use:
Linux, fedora 34

Which lsp-client do you use:
Vscode
Describe your project (alternative: link to the project):
stack based project

### Steps to reproduce
See steps gif:
![open-source-incorrect](https://user-images.githubusercontent.com/2716069/146671879-daf9d789-38da-4fff-8024-8022ff36e74f.gif)

### Expected behaviour
Correct page is opened on hackage
https://hackage.haskell.org/package/base-4.14.3.0/docs/src/Data-Foldable.html#concatMap

### Actual behaviour
Invalid URL is opened on hackage
https://hackage.haskell.org/package/base-4.14.3.0/docs/src/Data.Foldable.html#concatMap
`                                                                                      ^ It should be Data-Foldable, not Data.Foldable

### Include debug information
Execute in the root of your project the command `haskell-language-server-wrapper --debug .` and paste the logs here:
(if you are using the vscode extension check the executable location [here](https://github.com/haskell/vscode-haskell#downloaded-binaries))
<details>
<summary>
Debug output:
</summary>

```
<paste your logs here>
```
</details>

Paste the logs from the lsp-client, you can check instructions about for VS Code [here](https://github.com/haskell/vscode-haskell#troubleshooting)

<details>
<summary>
LSP logs:
</summary>

```
<paste your logs here>
```
</details>

**Comment by jhrcek** [comment]
This seems to be problem in vscode-haskell, which is replacing `-`s with `.`s here:
https://github.com/haskell/vscode-haskell/blob/7eb6a485e1e443c2eea195f0afe37e6b9963c22c/src/docsBrowser.ts#L160-L163

**Comment by jhrcek** [comment]
Looking around, I found that the **dots in haddock source documentation** are most likely result of this change in haddock: https://github.com/haskell/haddock/pull/977/files#diff-1ba3997884082b72ff50a289d8b53fa9df6948198e8494ffe7ffdf0044c19223R38

It seems to me that those dots started to appear as unintended consequence of changes in that PR. I'd like to check this with the author of that PR, but I can't tag him here. Anyone knows of a good person to ask?

**Comment by jneira** [comment]
Yeah, thanks for the report and the further investigation, moving to the extension repo
